node/lib/util/commit.js

```diff
@@ -550,6 +558,13 @@ exports.commit = co.wrap(function *(metaRepo,
             if (headCommit !== null) {
                 parents.push(headCommit);
             }
+
+            if (mergeTree !== null) {
+                const mergeSubParent = yield mergeTree.entryByPath(name);
+                if (mergeSubParent.id() !== headCommit.id()) {
+                    parents.push(mergeSubParent.id());
```

Comment:
in line 559, we just pushed a commit to `parents`, now we are pushing an object id to `parents`. Is it designed to have different types in the same array?

Reply:
It passes the tests, and it didn't when I accidentally pushed mergeSubParent (without the .id()).  So. apparently, yes.